src/WPF/ArcGISRuntime.WPF.Samples/Samples/Map/SearchPortalMaps/SearchPortalMaps.xaml.cs

```diff
@@ -0,0 +1,418 @@
+
+                // Get the user's content (items in the root folder and a collection of sub-folders)
+                PortalUserContent myContent = await portal.User.GetContentAsync();
+
+                // Get the web map items in the root folder
+                mapItems = from item in myContent.Items where item.Type == PortalItemType.WebMap select item;
+
+                // Loop through all sub-folders and get web map items, add them to the mapItems collection
+                foreach (PortalFolder folder in myContent.Folders)
+                {
+                    IEnumerable<PortalItem> folderItems = await portal.User.GetContentAsync(folder.FolderId);
+                    mapItems.Concat(from item in folderItems where item.Type == PortalItemType.WebMap select item);
+                }
+            }
+
+            // Show the web map portal items in the list box
+            MapListBox.ItemsSource = mapItems;
+        }
+
+        private void LoadMapButtonClick(object sender, RoutedEventArgs e)
+        {
+            // Get the selected web map item in the list box
+            PortalItem selectedMap = MapListBox.SelectedItem as PortalItem;
+            if (selectedMap == null) { return; }
+
+            // Create a new map, pass the web map portal item to the constructor
+            Map webMap = new Map(selectedMap);
+
+            // Show the web map in the map view
+            MyMapView.Map = webMap;
+        }
+
+        private void RadioButtonUnchecked(object sender, RoutedEventArgs e)
+        {
+            // When the search/user radio buttons are unchecked, clear the list box
+            MapListBox.ItemsSource = null;
+
+            // Set the map to the default (if necessary)
+            if (MyMapView.Map.Item != null)
+            {
+                DisplayDefaultMap();
+            }
+        }
+
+        private async Task<bool> EnsureLoggedInAsync()
+        {
+            bool loggedIn = false;
+
+            try
+            {
+                // Create a challenge request for portal credentials (OAuth credential request for arcgis.com)
+                CredentialRequestInfo challengeRequest = new CredentialRequestInfo();
+
+                // Use the OAuth implicit grant flow
+                challengeRequest.GenerateTokenOptions = new GenerateTokenOptions
+                {
+                    TokenAuthenticationType = TokenAuthenticationType.OAuthImplicit
+                };
+
+                // Indicate the url (portal) to authenticate with (ArcGIS Online)
+                challengeRequest.ServiceUri = new Uri(ArcGISOnlineUrl);
+
+                // Call GetCredentialAsync on the AuthenticationManager to invoke the challenge handler
+                var cred = await AuthenticationManager.Current.GetCredentialAsync(challengeRequest, false);
+                loggedIn = cred != null;
+            }
+            catch (OperationCanceledException ex)
+            {
+                // TODO: handle login canceled
```

Comment:
Can an actual message dialog/prompt be put in here to give the user information as to why the sample crashed.